File: src/frontend/src/lib/types/page.ts
import type { Category } from "./category";

export type PageRequest = {
    PageID: string;
};

export type Page = {
    backdropURL: string;
    id: string;
    name: string;
    overview: string;
    posterURL: string;
    categories: Category[];
};

export type Pages = {
    pageCount: number;
    pages: Page[];
};

File: src/go-backend/internal/http/model/page.go
package model

// Page Request
type PageRequest struct {
	PageID string
}

// Page Response
type Page struct {
	BackdropURL string     `json:"backdropURL"`
	ID          string     `json:"id"`
	Name        string     `json:"name"`
	Overview    string     `json:"overview"`
	PosterURL   string     `json:"posterURL"`
	Categories  []Category `json:"categories"`
}

// Pages Response
type Pages struct {
	PageCount int    `json:"pageCount"`
	Pages     []Page `json:"pages"`
}
